### PR TITLE
Time and data saving improvements

### LINKED ISSUE #64: Remove legacy requirements
Remove requirements no longer used like tensorflow

### PR DESCRIPTION
**Pull request by HanBnrd**
- [x] Fix duration start
- [x] Save audio as wav
- [x] ~Save spectrogram as csv~ (will be done with artefact viz)
- [x] Update requirements (Closes #64)
- [x] Fix Dobot AttributeError

### CODE REVIEW

File: modules/data_writer.py [hunk 1]
```diff
@@ -43,14 +44,17 @@ def json_update(self):
         self.data_file.write(json_object)
         self.data_file.write(',\n')
 
-    def terminate(self):
+    def terminate_data_writter(self):
```
Comment (craigvear) [inline modules/data_writer.py:47]: data_writer

File: modules/data_writer.py [hunk 2]
```diff
@@ -67,4 +71,4 @@ def writing_manager(self):
             self.json_update()
             sleep(0.1)
         logging.info("quitting data writer thread")
-        self.terminate()
+        self.terminate_data_writter()
```
Comment (craigvear) [inline modules/data_writer.py:74]: data_writer

**Review by craigvear — APPROVED** [review]

File: modules/data_writer.py (hunk 1 above)
Comment (HanBnrd) [reply modules/data_writer.py:47]: Good catch, thanks!

**Comment by HanBnrd** [comment]
PS: I used scipy and not librosa for the spectrogram because librosa's function doesn't like threads very much, and also scipy was already in requirements

**Comment by HanBnrd** [comment]
Will check with Adam if data formats are all good for him before merging this PR

**Comment by craigvear** [comment]
great - thank you!

**Comment by HanBnrd** [comment]
Hi @craigvear, ready for review and merge 😃

**Review by craigvear — APPROVED** [review]
Brill work Johann.

**Pull request merged**